### PR TITLE
Overhaul appointments reason and comments fields

### PR DESCRIPTION
**Pull request by JunTaoLuo**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- This PR primarily addresses the [ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/94339) for removing duplicate  logic for displaying comments on appointment details.
- After some discussion with the team, we felt that after moving the parsing logic to the BE, we no longer need to maintain complicated null state logic (e.g. falling back to "none" for CC appointments) and conjunction/partition logic (joining and splitting by ":") and instead rely on the `reasonForAppointment` and `patientComments` fields provided by the backend.
- Note that this change doesn't change the [behavior for handling v0 appointments](https://github.com/department-of-veterans-affairs/vets-website/blob/cc0e9917b4738afabae6d34ed09ab865b846fd36/src/applications/vaos/services/appointment/transformers.js#L216-L223) in the transformer.
- As a side effect of streamlining the logic, we also end up addressing three existing bugs:
  -  [92704](https://github.com/department-of-veterans-affairs/va.gov-team/issues/92704) is fixed because once we switch to the new logic there's no conjoining/splitting by ":" so colons in patient comments will automatically be preserved.
  - [94689](https://github.com/department-of-veterans-affairs/va.gov-team/issues/94689) is fixed because we no longer have null state logic fallback to "none" for CC appointments that's currently in `selectComment` and `selectAppointmentDetails`
  - I also found a bug where in CC appointments, if the patient comment contains a colon, only the content after the colon is displayed. This was caused by the layout component splitting the comment by colon when it shouldn't.
- Appointments (VAOS) Team
- This is not behind a Flipper toggle

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/94339
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/92704
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/94689

## Testing done

- Tested locally for the affected appointment types, see screenshots below
- Updated existing unit tests for the updated behaviors

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Colon in comments (fixed) |    image   |  image   |
| CC null state (fixed) |    image   | image |
| CC Details (bug fixed) |     image  |  image  |
| CC Request Details (other details only) |   image    |  image   |
| CC Request Cancellation (other details only) |   image    |   image  |
| VA Request Details (no change) |    image   |   image   |
| VA Request Cancellation (no change) |   image    |    image  |
| In Person Details (no change) |   image    |  image    |
| In Person Cancellation (no change) |    image   |   image   |
| Phone Details (no change) |   image    |   image   |
| Phone Cancellation (no change) |   image    |   image   |

## What areas of the site does it impact?

Appointments (VAOS)

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by CBonade — DISMISSED** [review]

**Review by jenniemc — CHANGES_REQUESTED** [review]
The overhaul looks good...just one comment about community care appointments. Because CC request flow does not ask for user's reason for appointment, it should not display "Reason: Not available" in the CC confirmed details page nor in the CC request details page. 

The [[figma design](https://www.figma.com/design/eonNJsp57eqfPqx7ydsJY9/Feature-Reference-%7C-Appointments-FE?node-id=1183-199238&node-type=section&t=9LhKYXoSKNX95R0l-0)] only displays "Other details" for CC

File: src/applications/vaos/components/layout/CCLayout.jsx [hunk 1]
```diff
@@ -37,7 +36,7 @@ export default function CCLayout({ data: appointment }) {
   if (!appointment) return null;
 
   const { address, providerName, treatmentSpecialty } = ccProvider;
-  const [reason, otherDetails] = comment ? comment?.split(':') : [];
+  const { reasonForAppointment, patientComments } = appointment || {};
```
[the remark this answers is not in the document]
Comment (JunTaoLuo) [reply src/applications/vaos/components/layout/CCLayout.jsx:39]: Great call! I keep forgetting the design was updated.

**Comment by JunTaoLuo** [comment]
CC appointment/request/cancellation all updated (code and screenshots).

**Review by jenniemc — APPROVED** [review]
LGTM!

**Review by simiadebowale — APPROVED** [review]

**Pull request merged**